### PR TITLE
[quant][core][feature] Implemented masked_fill for QuantizedCPU tensors

### PR DESCRIPTION
**Pull request by dzdang**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #78384
* __->__ #78368

Summary:
Previously, there was no suport for masked_fill for quantized tensors.
This PR introduces the feature for quantized CPU tensors. A
corresponding test case was added to test the impl.

Most of the implementation is copied over from the fp kernel and adapted
for quantized tensors.

TODO: implement masked_fill for QuantizedCUDA tensors.

Test Plan:
```
python test/test_quantization.py -k test_qtensor_masked_fill
```

Differential Revision: [D36706178](https://our.internmc.facebook.com/intern/diff/D36706178)

### CODE REVIEW

**Comment by dzdang** [comment]
@dzdang has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D36706178).

**Comment by dzdang** [comment]
@dzdang has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D36706178).

File: aten/src/ATen/native/TensorAdvancedIndexingUtils.h [hunk 1]
```diff
@@ -0,0 +1,93 @@
+/*
+The contents of this file are used in ATen/native/TensorAdvancedIndexing.cpp and ATen/native/quantized/TensorAdvancedIndexing.cpp
```
Comment (vkuzo) [inline aten/src/ATen/native/TensorAdvancedIndexingUtils.h:2]: nit: I'd remove this, it's easy for this to become outdated

Comment (dzdang) [reply aten/src/ATen/native/TensorAdvancedIndexingUtils.h:2]: will do

File: aten/src/ATen/native/quantized/cpu/kernels/QuantizedOpKernels.cpp [hunk 2]
```diff
@@ -3597,10 +3598,39 @@ void dequantize_tensor_per_tensor_affine_sub_byte_cpu(
...
+        *(scalar_t*)(dst + strides[0] * i) = quantize_val<scalar_t>(scale, zero_point, value);
+      }
+    }
+  };
+  iter.for_each(loop);
+}
+
+void masked_fill_kernel_quantized_cpu(TensorIterator& iter, const Scalar& value, double scale, int zero_point) {
+  AT_DISPATCH_QINT_TYPES(iter.dtype(), "masked_fill", [&] {
+    float scalar_val = value.to<float>();
```
[the remark this answers is not in the document]
Comment (dzdang) [reply aten/src/ATen/native/quantized/cpu/kernels/QuantizedOpKernels.cpp:3624]: good catch! I'll make that change

**Comment by dzdang** [comment]
@dzdang has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D36706178).

File: aten/src/ATen/native/TensorAdvancedIndexingUtils.h (hunk 1 above)
Comment (vkuzo) [reply aten/src/ATen/native/TensorAdvancedIndexingUtils.h:2]: would be good to resolve before landing

**Review by vkuzo — APPROVED** [review]

**Pull request closed without merging**